Resolves #226 "Transition from javax to jakarta"

rzo1 opened the pull request:
# What does this PR do?

This pull request brings compatibility to the Jakarta EE 9.1 Spec. Due to the namespace change this might break consumers, so I increased the version to 2.0.0-SNAPSHOT. 

Since Sweble is not maintained anymore and has a hard binding against an old JAXB, I decided to use a shading approach for the javax -> jakarta migration (like we do in numerous ASF projects with EE reference). 

During the upgrade I did several dependency upgrades or cleanups. For example, I removed the hard dependencies towards ant or trove4j. Either by forking (ant bzip2 code) or by replacing with stock JDK code. This might impact performance, but might still be better than relying on unmaintained libs. 

We ran the performance tests locally and the results are in #226, so in principle the code stack still works after the changes.

Happy to receive some feedback.

ukp-svc-jenkins commented:
Can one of the admins verify this patch?

rzo1 commented on de.tudarmstadt.ukp.wikipedia.mwdumper/src/main/java/de/tudarmstadt/ukp/wikipedia/mwdumper/shade/org/apache/tools/bzip2/BZip2Constants.java:
Might be worth to shade that in `deps` as well to remove code duplication and ease possible updates? wdyt?

reckart replied:
Do you have an idea why the code is copied anyway? Why not have an actual dependency?

rzo1 replied:
I don't think, that a library should rely on a build tool (here: ant) with all of it's stuff if we only need 5 classes related to bzip2 processing.

reckart replied:
Nothing related in commons-compress?

rzo1 replied:
I copied / forked it to discuss this issue within this PR. I would be in favour of only shading that 5 classes in `/deps` and re-use it over the code base.

rzo1 replied:
Let me check commons-compress

rzo1 replied:
looks good. removed forked code in favour of commons-compress.

mawiesne reviewed (approved):
Thanks for all the effort and the progress that this PR will bring.

rzo1 commented:
@reckart Can we reconfigure the Jenkins for this repo? The PR introduces GH actions, but it seems, that this repo still relies on a decomissioned (?) build instance ?

reckart commented:
I have removed the jenkins check from the master branch - is that what you were asking for?

rzo1 commented:
> I have removed the jenkins check from the master branch - is that what you were asking for?

Yes ;-)

reckart commented:
Hm, considering that the PR includes a github action - shouldn't this run automatically? Is there anything we need to do for it to activate?

rzo1 commented:
> Hm, considering that the PR includes a github action - shouldn't this run automatically? Is there anything we need to do for it to activate?

I think, that the GH action needs to be added on "main" first, I guess (or activated for the repo/org maybe?).

rzo1 commented:
Yeah, I have a similar impression. They are running on the fork: https://github.com/rzo1/dkpro-jwpl/actions/runs/6559026682/job/17813774348 - so an organization setting maybe?

reckart commented:
I'll tentatively approve to PR - let's see if that helps something...

reckart commented:
The repo settings say that first-time contributors need approval - but the PR already has two approvals now...

rzo1 commented:
Ok, now it is running ;-)

reckart commented:
Merged the other branch to main - let's see if this then also helps here... there are some conflicts now.

rzo1 commented:
Just rebased this PR, so should be ok now in terms of conflicts

reckart commented on de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/WikiHibernateUtil.java:
```diff
@@ -109,10 +112,10 @@ else if(jdbcURL.toLowerCase().contains("hsql")) {
 
         // Do only update schema on changes
-        if(useMySQL) {
+        if(useMySQL || useMariaDB) {
```
Having useMariaDB in both conditions seems pointless.

rzo1 replied:
Looked in the history. Looks like a refactoring error and should be hsqldb. I will fix it. Nice catch!

reckart commented on de.tudarmstadt.ukp.wikipedia.parser/pom.xml:
```diff
@@ -44,14 +44,14 @@ Please use the SWEBLE parser (http://sweble.org/) for parsing MediaWiki markup.
-		  <groupId>org.slf4j</groupId>
-		  <artifactId>slf4j-log4j12</artifactId>
-		  <scope>test</scope>
+		  <groupId>org.apache.logging.log4j</groupId>
```
Why provided instead of test?

rzo1 replied:
Think it is just CnP from my side.

It seems, that `parser` is intended as a library module. Other modules such as "dumper", etc. are more or less CLI modules (with provided log4j2 deps). Will fix it.

reckart commented on de.tudarmstadt.ukp.wikipedia.revisionmachine/src/main/java/de/tudarmstadt/ukp/wikipedia/revisionmachine/difftool/config/gui/panels/FilterPanel.java:
```diff
@@ -198,28 +198,28 @@ class FilterTableModel
-				{ "special(-1)", new Boolean(false), -1 },
-				{ "media(-2)", new Boolean(false), -2 }
+		private final Object[][] data = { { "main(0)", Boolean.FALSE, 0 },
+				{ "talk(1)", Boolean.FALSE, 1 },
```
Why not simply `true` and `false`?

reckart commented on de.tudarmstadt.ukp.wikipedia.timemachine/src/main/java/de/tudarmstadt/ukp/wikipedia/timemachine/dump/version/DumpVersionTroveIntKey.java:
```diff
@@ -45,29 +45,29 @@ public class DumpVersionTroveIntKey extends AbstractDumpVersion {
 	 * array list
 	 */
-	private TIntHashSet pageIdRevList;
+	private Set<Integer> pageIdRevList;
```
Wouldn't using something like FastUtil be reasonable here?

rzo1 replied:
Switched to fastutil-core

reckart commented on deps/pom.xml:
```diff
@@ -0,0 +1,33 @@
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>deps</artifactId>
```
IMHO artifact names should all share a common prefix like e.g. `jwpl`.

rzo1 replied:
I have no objections in changing to `dkpro-jwpl-XXX`. WDYT @mawiesne ?

reckart replied:
And while at it, the groupId could also be updated to `org.dkpro.jwpl`.

reckart replied:
(e.g. DKPro Core as groupId `org.dkpro.core` for some time now)

mawiesne replied:
see: https://github.com/dkpro/dkpro-jwpl/issues/165

mawiesne replied:
I agree. we could do the relocation / renaming as proposed in #165, but separately.

rzo1 replied:
Yeah - let's tackle this for 2.0.0 but in another PR to ease reviewer burden :)

reckart commented on pom.xml:
```diff
@@ -95,102 +95,58 @@
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-slf4j-impl</artifactId>
+        <version>${log4j2.version}</version>
+        <scope>runtime</scope>
```
Isn't runtime transitive? Do we want that? Normally, dependencies in dependency management should not have a scope.

mawiesne commented:
@reckart Could you please de-activate the Jenkins hook from this project's settings entirely? The other open PR is also blocked merely by Jenkins integration still interfering...

mawiesne commented:
> I have removed the jenkins check from the master branch - is that what you were asking for?

Please remove it project-wide, that is, from all branches / PRs. Thx in advance.

rzo1 commented:
@reckart Addressed most of your comments. I would like to do the renaming of artifacts + group-ids in https://github.com/dkpro/dkpro-jwpl/issues/165 after this PR is merged.

reckart commented:
> Please remove it project-wide, that is, from all branches / PRs. Thx in advance.

It is removed, but if there are old PRs that still have a failed state from Jenkins, I'm afraid GitHub doesn't seem to be smart enough to remove those... ping me in such cases, I can force-merge them.

reckart commented:
@rzo1 So, are we good to merge?

reckart commented:
@rzo1 then feel free to hit the button
